src/Symfony/Component/Form/Extension/Core/Type/MultiStepType.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Silas Joisten <silasjoisten@proton.me>
+ * @author Patrick Reimers <preimers@pm.me>
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+final class MultiStepType extends AbstractType
+{
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setRequired('steps')
+            ->setDefault('current_step', static function (Options $options): string {
+                /** @var array<string, mixed> $steps */
+                $steps = $options['steps'];
+                $firstStep = array_key_first($steps);
+                if (!\is_string($firstStep)) {
+                    throw new \InvalidArgumentException('The option "steps" must be an associative array.');
```

Comment:
This validation process should be part of the `steps` option and should be implemented within the `setAllowedValues()` method for that option, otherwise the validation will be bypassed when the user sets the `current_step` value.